src/OrbitGl/MemoryTrack.cpp

```diff
@@ -10,91 +10,44 @@
-  if (value_lower_bound_.has_value()) {
-    std::string text = value_lower_bound_.value().first;
-    float string_width = text_renderer.GetStringWidth(text.c_str(), font_size);
-    Vec2 text_box_size(string_width, layout_->GetTextBoxHeight());
-    Vec2 text_box_position(pos_[0] + size_[0] - text_box_size[0] - layout_->GetRightMargin() -
-                               layout_->GetSliderWidth(),
-                           content_pos[1] - content_height);
-    text_renderer.AddText(text.c_str(), text_box_position[0],
-                          text_box_position[1] + layout_->GetTextOffset(), text_z, kWhite,
-                          font_size, text_box_size[0]);
-  }
+template <size_t Size>
+void MemoryTrack<Size>::Draw(Batcher& batcher, TextRenderer& text_renderer,
+                             uint64_t current_mouse_time_ns, PickingMode picking_mode,
+                             float z_offset) {
+  GraphTrack<Size>::Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode, z_offset);
+  AnnotationTrack::DrawAnnotation(batcher, text_renderer, picking_mode,
+                                  GlCanvas::kZValueTrackText + z_offset);
 }
 
-void MemoryTrack::SetWarningThresholdWhenEmpty(const std::string& pretty_label, double raw_value) {
-  if (warning_threshold_.has_value()) return;
-  warning_threshold_ = std::make_pair(pretty_label, raw_value);
+template <size_t Size>
+void MemoryTrack<Size>::SetWarningThresholdWhenEmpty(const std::string& pretty_label,
+                                                     double raw_value) {
+  AnnotationTrack::SetWarningThresholdWhenEmpty(pretty_label, raw_value);
   UpdateMinAndMax(raw_value);
 }
 
-void MemoryTrack::SetValueUpperBoundWhenEmpty(const std::string& pretty_label, double raw_value) {
-  if (value_upper_bound_.has_value()) return;
-  value_upper_bound_ = std::make_pair(pretty_label, raw_value);
+template <size_t Size>
+void MemoryTrack<Size>::SetValueUpperBoundWhenEmpty(const std::string& pretty_label,
+                                                    double raw_value) {
+  AnnotationTrack::SetValueUpperBoundWhenEmpty(pretty_label, raw_value);
   UpdateMinAndMax(raw_value);
 }
-void MemoryTrack::SetValueLowerBoundWhenEmpty(const std::string& pretty_label, double raw_value) {
-  if (value_lower_bound_.has_value()) return;
-  value_lower_bound_ = std::make_pair(pretty_label, raw_value);
+
+template <size_t Size>
+void MemoryTrack<Size>::SetValueLowerBoundWhenEmpty(const std::string& pretty_label,
+                                                    double raw_value) {
+  AnnotationTrack::SetValueLowerBoundWhenEmpty(pretty_label, raw_value);
   UpdateMinAndMax(raw_value);
 }
 
-void MemoryTrack::UpdateMinAndMax(double value) {
-  max_ = std::max(max_, value);
-  min_ = std::min(min_, value);
+template <size_t Size>
+void MemoryTrack<Size>::UpdateMinAndMax(double value) {
+  this->max_ = std::max(this->max_, value);
+  this->min_ = std::min(this->min_, value);
```

Comment:
This is because we are inherit from the template class `GraphTrack<Size>`, so the member variable name depends on the template variable `Size`. We need either:
```
using GraphTrack<Size>::min_;
using GraphTrack<Size>::max_;
```
or use `this->min_` and `this->max` alternatively to help the complier find the definition of the name at the right place.

Reply:
`this->min_` and `this->max_` is the more common approach. So +1 for this solution from my side.

Reply:
I did not know about that. Thank you for the explanation!